### PR TITLE
Allow requiring a secret for incoming JIRA webhook requests

### PR DESCRIPTION
**Pull request by valentijnscholten**
The current JIRA webhook can be used just be sending POST data to `/webhook`. 
This PR changes this into `/jira/webhook/<secret>` where secret is a slug to be configured in the system settings.
For backwards compatibility reasons we keep the `/webhook/` url working as well.

Note:
- For existing installs the JIRA Webhook secret will be disabled to keep them working
- For new installs the webhook is disabled anyway, but the initializer generates a random (uuid) secret and enabled secret by default;

System Settings page now looks like this:

image

The disable option is mainly to keep existing installs from breaking on upgrade.

Docs: https://github.com/DefectDojo/Documentation/pull/145

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

**Review by madchap — CHANGES_REQUESTED** [review]
I am going to nitpick... but I'd love to have a small "copy" icon next to that link :)

File: dojo/jira_link/views.py [hunk 1]
```diff
@@ -26,12 +26,23 @@
 
 
 @csrf_exempt
-def webhook(request):
-    # Webhook shouldn't be active if jira isn't enabled
+def webhook(request, secret=None):
```
Comment (madchap) [inline dojo/jira_link/views.py:29]: `@require_http_methods(["POST"])` maybe? What do we want to allow through that webhook?

Comment (valentijnscholten) [reply dojo/jira_link/views.py:31]: add. good idea, although it makes testing it harder ;-)

**Review by madchap — APPROVED** [review]

**Comment by valentijnscholten** [comment]
copy button would be nice, but takes some time and can be added by anyone who loves javascript ;-)

**Pull request merged**